modin/core/io/text/csv_glob_dispatcher.py

```diff
@@ -348,14 +348,27 @@ def file_exists(cls, file_path: str, storage_options=None) -> bool:
                 else:
                     new_storage_options = {}
 
-                s3fs = S3FS.S3FileSystem(anon=False, **new_storage_options)
+                fs, _ = fsspec.core.url_to_fs(file_path, **new_storage_options)
                 exists = False
                 try:
-                    exists = len(s3fs.glob(file_path)) > 0 or exists
-                except (NoCredentialsError, PermissionError, EndpointConnectionError):
-                    pass
-                s3fs = S3FS.S3FileSystem(anon=True, **new_storage_options)
-                return exists or len(s3fs.glob(file_path)) > 0
+                    exists = fs.exists(file_path)
```

Comment:
the `try` block is checking `fs.exists(file_path)`, while the except block is checking `len(fs.glob(file_path)) > 0`. Both blocks will check `len(fs.glob(file_path)) > 0` if their respective check fails. Could you make the `try` and `except` branches check the same things (in the original code they are both just checking `len(s3fs.glob(file_path)) > 0`).

Reply:
I've aligned try-except block (now both check using `fs.exists`). `exists` is faster than `fs.glob` because it just check availability of `file_path`. On `return` stage I think it's better to use `fs.glob` in case `exists` is False (leave return phase as it is).

Reply:
Okay, but still we check `len(fs.glob(file_path)) > 0` twice if it's false and we hit the `except`? Is there a good way to avoid that? It's confusing to read. 

I know this problem was there before your PR...

Reply:
I think that we should leave this logic with trying to check with `exists()` (and in case it's `False` use `fs.glob`). The reason of check with `fs.glob` is addresses like `s3://url_path/*`. In this case `exists` returns `False`, but `fs.glob` returns all files placed in  s3://url_path/